### PR TITLE
feat(cloudflare): add local mocking for D1 bindings

### PR DESCRIPTION
**Pull request by alexanderniebuhr**
## Changes

- **new:** local mocking for D1 bindings

## Testing
- existing tests
- added tests for D1

## Docs
/cc @withastro/maintainers-docs for feedback!

### CODE REVIEW

**Comment by alexanderniebuhr** [comment]
Current dependencies on/for this PR:
* main
  * **PR #8654** <a href="https://app.graphite.dev/github/pr/withastro/astro/8654" target="_blank">Graphite</a> 
    * **PR #8655** <a href="https://app.graphite.dev/github/pr/withastro/astro/8655" target="_blank">Graphite</a>  👈
      * **PR #8656** <a href="https://app.graphite.dev/github/pr/withastro/astro/8656" target="_blank">Graphite</a> 
        * **PR #8657** <a href="https://app.graphite.dev/github/pr/withastro/astro/8657" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/withastro/astro/8655?utm_source=stack-comment).

**Review by matthewp — APPROVED** [review]

File: packages/integrations/cloudflare/README.md [hunk 1]
```diff
@@ -212,7 +212,11 @@ export default defineConfig({
 
 ## Cloudflare runtime
 
-Gives you access to [environment variables](https://developers.cloudflare.com/pages/platform/functions/bindings/#environment-variables).
+Gives you access to [environment variables](https://developers.cloudflare.com/pages/platform/functions/bindings/#environment-variables), [Cloudflare bindings](https://developers.cloudflare.com/pages/platform/functions/bindings).
```
Comment (sarah11918) [inline packages/integrations/cloudflare/README.md:215]:
```suggestion
Gives you access to [environment variables](https://developers.cloudflare.com/pages/platform/functions/bindings/#environment-variables), and [Cloudflare bindings](https://developers.cloudflare.com/pages/platform/functions/bindings).
```

File: .changeset/hungry-mails-boil.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@astrojs/cloudflare': minor
+---
+
+Introduce support for local D1 bindings. Enhance development experience by allowing direct integration with `astro dev`.
```
Comment (sarah11918) [inline .changeset/hungry-mails-boil.md:5]:
```suggestion
Introduces support for local D1 bindings. Enhances development experience by allowing direct integration with `astro dev`.
```
Only an example, totally not a required change, but my *personal* preferred verb tense for these statements is the present tense (not imperative).

And if not full sentences on their own, choose the tense as if it completed the phrase, "This update... " or "The changes in this update...."

e.g. "This update introduces support for local D1 bindings..." If you wanted a standard way of thinking of it!

**Review by sarah11918 — APPROVED** [review]
Looks good to me! Just an idea for the changeset verb tense, in case you felt you wanted guidance, and one little missing "and" below!  :partying_face:

**Pull request merged**